run-cleanup.sh

```diff
@@ -2,49 +2,71 @@
 
 set -euo pipefail
 
-if [[ $# -ne 2 ]]; then
-  echo "Invalid arguments. Use: $0 SRC DST"
+if [[ $# -ne 3 ]]; then
+  echo "Invalid arguments. Use: $0 SRC TMP DST"
   exit 1
 fi
 
 SRC_REPOSITORY="$1"
-CLEANED_REPOSITORY="$2"
-shift 2
+ORDERED_REPOSITORY="$2"
+CLEANED_REPOSITORY="$3"
+shift 3
 
 if [[ ! -d "${SRC_REPOSITORY}" ]]; then
   echo "Invalid argument. ${SRC_REPOSITORY} has to be a directory."
   exit 1
 fi
 
+if [[ -e "${ORDERED_REPOSITORY}" ]]; then
+  echo "Invalid argument. ${ORDERED_REPOSITORY} may not exist."
+  exit 1
+fi
+
 if [[ -e "${CLEANED_REPOSITORY}" ]]; then
   echo "Invalid argument. ${CLEANED_REPOSITORY} may not exist."
   exit 1
 fi
 
 
 BASE="$(dirname "$(readlink -f "$0")")"
-SETUP_CLEANUP="${BASE}/setup-cleanup.sh"
+SETUP_MERCURIAL="${BASE}/setup-mercurial.sh"
 VIRTUALENV="${BASE}/data/py3-env"
 
-if ! /bin/bash "${SETUP_CLEANUP}"; then
+if ! /bin/bash "${SETUP_MERCURIAL}"; then
   echo "Error during setup."
   exit 2
 fi
 source "${VIRTUALENV}/bin/activate"
-
 # Disable all extensions.
 # (https://stackoverflow.com/questions/46612210/mercurial-disable-all-the-extensions-from-the-command-line)
-HGRCPATH= hg \
+export HGRCPATH=
+export HGPLAIN=
+
+
+echo "Cloning official repository"
+hg clone "http://hg.fast-downward.org" "${ORDERED_REPOSITORY}"
+
+if hg -R "${SRC_REPOSITORY}" incoming "${ORDERED_REPOSITORY}"; then
```

Comment:
Is this test still necessary? After the pull in line 56 the condition will be satisfied in the new repository, even if it wasn't before. I think I suggested this test but this was when we were still removing parts from the official clone before pulling. 

I don't see how removing this test would cause any problems.

Reply:
I think this test is still necessary. It enforces that all current changes from the last Mercurial master are pulled in by the user prior to executing our script. Any 'multiple head on the same branch' issues (or other issues the pull could cause and we do not know about) can be seen in the user's repository.

Reply:
> It enforces that all current changes from the last Mercurial master are pulled in by the user prior to executing our script.

I don't see why that would be necessary. The next step after this test is a pull and after this pull the condition that we test here is satisfied even if it wasn't before.

If our assumption is that the repository does not have multiple heads on a single branch, we should test that assumption explicitly instead of testing something unrelated that would force the user to do a step that generates a warning (not an error) when the assumption is not satisfied.
(Here is an example for testing this but note the comment about the test in line #19: https://gist.github.com/FernFerret/3178035)

Reply:
Our problem is, we do not exactly know when fast-export succeeds and when it fails. Malte tested it and said there are cases when multiple heads on the same branch work. We simply do not know for what exactly we have to test (and additionally, what else we do not know what else could fast-export to fail). We do not want to say you are not allowed to convert this repository, although it would convert perfectly fine.

therefore, we kept the check that all changes have to be pulled. If fast-export tells you know "you have multiple heads on branch X" or "fix that". Those issues are in the user's repository and he can fix them there. 
If we would pull the newest changes ourselves, then 1. he could not find the problems that caused the error in his repository and 2. could not fix them.
(yes, he could find and fix them by pulling himself, but for me it is more naturally: "I get an error, I look into my repository" and not: "I  get an error, let's look through all the temporary files")

Reply:
I think fast-export doesn't produce an error in this case, only a warning and an incompatible repository, But if you already discussed this, I'm fine with leaving things as they are.